src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMutation.cs

```diff
@@ -228,4 +228,33 @@ public async Task<FormulaValue> InvokeAsync(FormulaType irContext, FormulaValue[
             return new InMemoryTableValue(IRContext.NotInSource(tableValue.Type), resultRows);
         }
     }
+
+    // Clear(collection_or_table)
+    internal class ClearImpl : ClearFunction, IAsyncTexlFunction3
+    {
+        public async Task<FormulaValue> InvokeAsync(FormulaType irContext, FormulaValue[] args, CancellationToken cancellationToken)
+        {
+            if (args[0] is ErrorValue errorValue)
+            {
+                return errorValue;
+            }
+
+            if (args[0] is BlankValue)
+            {
+                return irContext == FormulaType.Void ? FormulaValue.NewVoid() : FormulaValue.NewBlank(FormulaType.Boolean);
```

Comment:
>FormulaType.Void  [](http://example.com/codeflow?start=36&length=17)

This raises an question  - how do voids propagate errors today?

Reply:
For in memory collections, the only way for `Clear` function to return an error is if the argument is an `ErrorValue`. Other implementations (like PFx.Dataverse) will handle differently.